core/src/main/java/bisq/core/offer/OpenOfferManager.java

```diff
@@ -492,14 +495,29 @@ private void onRemoved(@NotNull OpenOffer openOffer, ResultHandler resultHandler
     // Close openOffer after deposit published
     public void closeOpenOffer(Offer offer) {
         getOpenOfferById(offer.getId()).ifPresent(openOffer -> {
-            openOffers.remove(openOffer);
+            removeFromOpenOffers(openOffer);
             openOffer.setState(OpenOffer.State.CLOSED);
             offerBookService.removeOffer(openOffer.getOffer().getOfferPayload(),
                     () -> log.trace("Successful removed offer"),
                     log::error);
         });
     }
 
+    private void removeFromOpenOffers(OpenOffer offer) {
+        openOffers.remove(offer);
+
+        // extract my node address
+        NodeAddress myNodeAddress = offer.getOffer().getOfferPayload().getOwnerNodeAddress();
+
+        // check
+        if(openOffers.stream().noneMatch(openOffer -> openOffer.getOffer().getOfferPayload().getOwnerNodeAddress().equals(myNodeAddress))) {
+            // check ongoing trades
+            if(tradeManager.getTradableList().stream().noneMatch(trade -> trade.getContract().getBuyerNodeAddress().equals(myNodeAddress) || trade.getContract().getSellerNodeAddress().equals(myNodeAddress)))
+                        p2PService.reportUnusedNodeAddress(myNodeAddress);
```

Comment:
ad missing role check: well, it is not that easy to determine what the clients role is at that point. Your code suggestion only works for trades you created. It fails if you took an offer from someone else. If one is doing the check thoroughly, it is much more complex, also because necessary info may not be provided (null) in certain cases. The approach now is to add any and all onion addresses that are anywhere related to the cause and prevent them from being deleted. Plus, even if a HS is wrongly kept once, it may be released in the future without any further calls and checks and complexity in the code.

ad null-check: fixed.

ad rename: I haven't touched the names as there is no Javadoc that explains what is what.

Reply:
Ah, yes you are right, OpenOfferManager handles it differently than tradeManager. But for OpenOffers my suggestion is correct as you are the maker. But for trades we need to use the pubKey to detect which role is ours. A util method in Contract ight be good as that is used anyway at other pleces as well (arbitration as far I remember).